lib/spack/spack/cmd/mirror.py

```diff
@@ -190,18 +190,22 @@ def mirror_create(args):
                 new_specs.add(s)
         specs = list(new_specs)
 
+    # Skip external specs, as they are already installed
+    external_specs = [s for s in specs if s.external]
+    specs = [s for s in specs if not s.external]
+
+    for spec in external_specs:
+        msg = 'Skipping {0} as it is an external spec.'
+        tty.msg(msg.format(spec.cshort_spec))
+
     # Default name for directory is spack-mirror-<DATESTAMP>
     directory = args.directory
     if not directory:
         timestamp = datetime.now().strftime("%Y-%m-%d")
         directory = 'spack-mirror-' + timestamp
 
     # Make sure nothing is in the way.
-    existed = False
-    if os.path.isfile(directory):
-        tty.error("%s already exists and is a file." % directory)
-    elif os.path.isdir(directory):
-        existed = True
+    existed = True if os.path.isdir(directory) else False
```

Comment:
@alalazo why remove the check for the path existing but being a file? Is this anticipated to be uncommon? Or am I missing somewhere this is guaranteed to be checked already? If the path exists but is a file wouldn't the next statements fail?

Otherwise this can just be `existed = bool(os.path.isdir(directory))`

Reply:
> Or am I missing somewhere this is guaranteed to be checked already?

The next function call to:
```python
present, mirrored, error = spack.mirror.create(directory, specs, num_versions=args.one_version_per_spec)
```
duplicates these same lines:

https://github.com/spack/spack/blob/e9a3e3bfbb17bc42073be0c5e6d7b9efc476d9f7/lib/spack/spack/mirror.py#L176-L178

> Otherwise this can just be existed = bool(os.path.isdir(directory))

Good point. I went with the even shorter:
```python
existed = os.path.isdir(directory)
```